HHH-12220 Feature/spatial restriction geometrytype

ridoo opened the pull request:
Currently the `st_geomtrytype` function is supported by HQL only. For those projects using the Criteria API it would be useful to have this spatial function available as well.

vladmihalcea commented:
@ridoo I can't integrate your Pull Request because it has Merge commits.

You can easily fix it by creating a new branch: HHH-12220, and --squash during merge like this:

    git merge --squash feature/spatial-restriction-geometrytype

Push the new HHH-12220 branch which, ideally, should contain a single commit.

If you didn't have Merge commits, I could cherry-pick the changes, but that's not possible now.

maesenka commented on hibernate-spatial/src/main/java/org/hibernate/spatial/SpatialDialect.java:
```diff
@@ -82,6 +82,24 @@
+	 * 
+	 * @return the geometry type name
+	 */
+	public String getGeometryTypeValue(GeometryType geometryType);
```
I find this method name rather confusing. I think getNameFor(GeometryType) would be clearer.

ridoo commented:
I've added some tests .. but to be honest I run tests only for postgresql for now. IMHO it is not very intuitive to run tests for each database ..

vladmihalcea commented:
@ridoo Can you create a separate branch, then `git merge --squash` all your changes from this branch and open a new Pull Request.

It's difficult to rebase this branch since it has merge commits.

Thanks.

ridoo commented:
@vladmihalcea I updated `master` and squashed made changes into a fresh branch ... after resolving some merge conflicts, there is a dependency which cannot be resolved. Gradle says
```
Successfully started process 'Gradle Compiler Daemon 1'
Started Gradle compiler daemon (0.55 secs) with fork options DaemonForkOptions{minHeapSize=null, maxHeapSize=null, jvmArgs=[], classpath=[/usr/share/java/groovy-all-2.x.jar, /usr/share/java/ant-1.9.6.jar, /usr/share/java/ant-launcher-1.9.6.jar]}.
Executing org.gradle.api.internal.tasks.compile.ApiGroovyCompiler@36dd4b10 in compiler daemon.
startup failed:
/home/henning/data/coding/java/hibernate-orm/buildSrc/src/main/groovy/org/hibernate/build/HibernateBuildPlugin.groovy: 21: unable to resolve class org.hibernate.build.gradle.testing.matrix.MatrixTestingPlugin
Exception executing org.gradle.api.internal.tasks.compile.ApiGroovyCompiler@36dd4b10 in compiler daemon: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details..
 @ line 21, column 1.
   import org.hibernate.build.gradle.testing.matrix.MatrixTestingPlugin
   ^

1 error

:buildSrc:compileGroovy FAILED
:compileGroovy (Thread[main,5,main]) completed. Took 1.501 secs.
Stopped 1 compiler daemon(s).

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':compileGroovy'.
> Compilation failed; see the compiler error output for details.

* Try:
Run with --stacktrace option to get the stack trace. Run with --debug option to get more log output.

BUILD FAILED

Total time: 4.56 secs
```

For what I can see, however, is, that the dependency is [where it should be](http://snapshots.jboss.org/maven2/org/hibernate/build/gradle/hibernate-matrix-testing/2.0.0-SNAPSHOT/), as [build.gradle](https://github.com/hibernate/hibernate-orm/blob/master/buildSrc/build.gradle#L17) adds the proper SNAPSHOT repo.

I (forced) pushed to https://github.com/ridoo/hibernate-orm/tree/HHH-12220

Any idea?

vladmihalcea commented:
@ridoo I still see 15 commits and Merge branch commits. Try to squash them all in a single commit as I indicated previously and then:

    git push origin feature/spatial-restriction-geometrytype -f

ridoo commented:
@vladmihalcea  I pushed the squashed commit from (https://github.com/ridoo/hibernate-orm/tree/HHH-12220)

vladmihalcea commented:
@ridoo Great, thanks.

vladmihalcea commented:
Superseded by [this PR](https://github.com/hibernate/hibernate-orm/pull/2234).
